src/mscorlib/src/System/Reflection/RuntimePropertyInfo.cs

```diff
@@ -25,8 +26,40 @@ internal unsafe sealed class RuntimePropertyInfo : PropertyInfo
         private BindingFlags m_bindingFlags;
         private Signature m_signature;
         private ParameterInfo[] m_parameters;
+        private volatile RuntimePropertyFlags m_lazyRuntimePropertyFlags;
+
+        [Flags]
+        private enum RuntimePropertyFlags
+        {
+            Computed = 0x00000001,
+            IsByRef = 0x00000002,
+            IsReadOnlyByRef = 0x00000004,
+        }
         #endregion
 
+        private RuntimePropertyFlags Flags
+        {
+            get
+            {
+                RuntimePropertyFlags flags = m_lazyRuntimePropertyFlags;
+                if ((flags & RuntimePropertyFlags.Computed) == 0)
+                {
+                    flags = RuntimePropertyFlags.Computed;
+                    if (PropertyType.IsByRef)
+                    {
+                        flags |= RuntimePropertyFlags.IsByRef;
+
+                        if (IsDefined(typeof(IsReadOnlyAttribute), inherit: false))
```

Comment:
`IsReadOnlyAttribute` does not have unique identity. C# compiler will create duplicates as necessary. E.g. try compiling against netstandard1.x.